src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3StreamTests.cs

```diff
@@ -2516,5 +2516,90 @@ public async Task PostRequest_ServerReadsPartialAndFinishes_SendsBodyWithEndStre
             // TODO(JamesNK): Check for abort of request side of stream after https://github.com/dotnet/aspnetcore/issues/31970
             Assert.Contains(LogMessages, m => m.Message.Contains("the application completed without reading the entire request body."));
         }
+
+        [Fact]
+        public async Task HEADERS_WriteLargeResponseHeaderSection_Success()
```

Comment:
Add a test for the response headers too large error condition?

Reply:
What do you mean by too large? There are existing tests that check to see whether individual headers are too large (a Kestrel configuration limit) or the header section has exceeded the limit specified by peer settings.

Reply:
I meant over the peer limit, but if that's already covered then great.